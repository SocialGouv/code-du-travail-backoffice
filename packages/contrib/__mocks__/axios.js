import "axios";

jest.mock("axios", () => {
  const axios = {
    interceptors: {
      response: {
        use: () => void 0
      }
    },

    delete: async () => void 0,
    get: async () => void 0,
    patch: async () => void 0,
    post: async () => void 0,
    put: async () => void 0
  };

  jest.spyOn(axios, "delete");
  jest.spyOn(axios, "get");
  jest.spyOn(axios, "patch");
  jest.spyOn(axios, "post");
  jest.spyOn(axios, "put");

  return {
    ...axios,
    create: () => axios
  };
});
