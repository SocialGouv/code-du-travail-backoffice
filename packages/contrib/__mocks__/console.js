/* eslint-disable no-undef */

global.console = {
  debug: jest.fn(),
  error: jest.fn(),
  info: jest.fn(),
  log: jest.fn(),
  warn: jest.fn()
};
