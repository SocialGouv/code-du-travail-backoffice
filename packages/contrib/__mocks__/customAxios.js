import "../src/libs/customAxios";

jest.mock("../src/libs/customAxios", () => () => {
  const customAxios = {
    delete: async () => void 0,
    get: async () => void 0,
    patch: async () => void 0,
    post: async () => void 0,
    put: async () => void 0
  };

  jest.spyOn(customAxios, "delete");
  jest.spyOn(customAxios, "get");
  jest.spyOn(customAxios, "patch");
  jest.spyOn(customAxios, "post");
  jest.spyOn(customAxios, "put");

  return customAxios;
});
