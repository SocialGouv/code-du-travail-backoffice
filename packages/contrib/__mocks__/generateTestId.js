global.generateTestId = require("uuid/v4");
