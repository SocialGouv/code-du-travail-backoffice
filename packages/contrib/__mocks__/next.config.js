module.exports = {
  serverRuntimeConfig: {},
  publicRuntimeConfig: {
    API_URI: "http://localhost:3200"
  }
};
