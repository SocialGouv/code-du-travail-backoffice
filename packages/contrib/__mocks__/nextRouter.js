import Router from "next/router";

Router.router = {
  push: () => {},
  prefetch: () => {}
};
