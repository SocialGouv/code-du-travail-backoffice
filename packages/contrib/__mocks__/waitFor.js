global.waitFor = delayInMs =>
  new Promise(resolve => setTimeout(resolve, delayInMs));
