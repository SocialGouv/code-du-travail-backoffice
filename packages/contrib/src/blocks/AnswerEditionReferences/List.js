import styled from "@emotion/styled";
import React from "react";
import { Flex } from "rebass";

import _Button from "../../elements/Button";
import Icon from "../../elements/Icon";

const Container = styled(Flex)`
  border-left: solid 1px var(--color-light-steel-blue);
  margin-left: 1rem;
  padding: 0 1rem;
`;

const Text = styled.span`
  margin-left: 1rem;
`;
const Button = styled(_Button)`
  padding: 0 0 0 0.5rem;
`;

export default class AnswerEditionReferencesBlockList extends React.PureComponent {
  openUrl(url) {
    window.open(url, "_blank");
  }

  render() {
    const { ariaName, onRemove, references } = this.props;
    const urlButtonTitle = `Bouton ouvrant le lien associé à ${ariaName}`;

    return (
      <Container flexDirection="column">
        {references.map(({ url, value }, index) => (
          <Flex alignItems="center" key={index}>
            <Icon color="black" icon="circle" isSmall />
            <Text>{value}</Text>
            {typeof url === "string" && url.length !== 0 && (
              <Button
                icon="link"
                onClick={() => this.openUrl(url)}
                title={`${urlButtonTitle} "${value}"`}
              />
            )}
            <Button
              icon="trash"
              onClick={() => onRemove(value)}
              title={`Bouton supprimant ${ariaName} "${value}"`}
            />
          </Flex>
        ))}
      </Container>
    );
  }
}
