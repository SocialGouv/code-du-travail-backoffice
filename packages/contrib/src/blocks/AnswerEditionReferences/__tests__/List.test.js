import React from "react";
import { fireEvent, render } from "@testing-library/react";

import "../../../../__mocks__/waitFor";

global.open = jest.fn();

import List from "../List";

describe.skip("[Contrib] blocks/AnswerEditionReferences/<List />", () => {
  const props = {
    ariaName: "A Reference Type",
    onRemove: jest.fn(),
    references: [
      {
        category: "labor_code",
        url: null,
        value: "L1234-5"
      },
      {
        category: null,
        url: "https://example.com",
        value: "An Uncategorized Reference"
      },
      {
        category: "agreement",
        url: null,
        value: "An Agreement Reference"
      }
    ]
  };

  const γ = render(<List {...props} />);

  it("should match snapshot", () => {
    expect(γ.container).toMatchSnapshot();
    expect(γ.queryByText(props.references[0].value)).toBeInTheDocument();
    expect(γ.queryByText(props.references[1].value)).toBeInTheDocument();
    expect(γ.queryByText(props.references[2].value)).toBeInTheDocument();
  });

  it("should call `onRemove()` prop with the expected param", async () => {
    fireEvent.click(
      γ.getByTitle(
        `Bouton supprimant ${props.ariaName} "${props.references[0].value}"`
      )
    );

    expect(props.onRemove).toHaveBeenCalledWith(props.references[0].value);
  });

  it("should call `window.open()` with the expected params", async () => {
    fireEvent.click(
      γ.getByTitle(
        `Bouton ouvrant le lien associé à ${props.ariaName} ` +
          `"${props.references[1].value}"`
      )
    );

    expect(global.open).toHaveBeenCalledWith(props.references[1].url, "_blank");
  });
});
