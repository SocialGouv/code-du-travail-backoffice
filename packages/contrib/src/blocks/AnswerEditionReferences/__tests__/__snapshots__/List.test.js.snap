// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[Contrib] blocks/AnswerEditionReferences/<List /> should match snapshot 1`] = `
<div>
  <div
    class="List__Container-sc-1pm6mpm-0 eKoGxR css-l3rx45"
  >
    <div
      class="css-zkfaav"
    >
      <svg
        aria-hidden="true"
        class="svg-inline--fa fa-circle fa-w-16 fa-sm Icon-sc-4jlog9-0 gomlef"
        color="black"
        data-icon="circle"
        data-prefix="fas"
        focusable="false"
        role="img"
        viewBox="0 0 512 512"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M256 8C119 8 8 119 8 256s111 248 248 248 248-111 248-248S393 8 256 8z"
          fill="currentColor"
        />
      </svg>
      <span
        class="List__Text-sc-1pm6mpm-1 bKJbSY"
      >
        L1234-5
      </span>
      <button
        class="Button-sc-1bpx84u-0 eyWUJN css-vhv7r1"
        title="Bouton supprimant A Reference Type \\"L1234-5\\""
      >
        <svg
          aria-hidden="true"
          class="svg-inline--fa fa-trash fa-w-14 fa-1x Icon-sc-4jlog9-0 hXLKsx"
          color="var(--color-lapis-lazuli)"
          data-icon="trash"
          data-prefix="fas"
          focusable="false"
          role="img"
          viewBox="0 0 448 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M432 32H312l-9.4-18.7A24 24 0 0 0 281.1 0H166.8a23.72 23.72 0 0 0-21.4 13.3L136 32H16A16 16 0 0 0 0 48v32a16 16 0 0 0 16 16h416a16 16 0 0 0 16-16V48a16 16 0 0 0-16-16zM53.2 467a48 48 0 0 0 47.9 45h245.8a48 48 0 0 0 47.9-45L416 128H32z"
            fill="currentColor"
          />
        </svg>
      </button>
    </div>
    <div
      class="css-zkfaav"
    >
      <svg
        aria-hidden="true"
        class="svg-inline--fa fa-circle fa-w-16 fa-sm Icon-sc-4jlog9-0 gomlef"
        color="black"
        data-icon="circle"
        data-prefix="fas"
        focusable="false"
        role="img"
        viewBox="0 0 512 512"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M256 8C119 8 8 119 8 256s111 248 248 248 248-111 248-248S393 8 256 8z"
          fill="currentColor"
        />
      </svg>
      <span
        class="List__Text-sc-1pm6mpm-1 bKJbSY"
      >
        An Uncategorized Reference
      </span>
      <button
        class="Button-sc-1bpx84u-0 eyWUJN css-vhv7r1"
        title="Bouton ouvrant le lien associé à A Reference Type \\"An Uncategorized Reference\\""
      >
        <svg
          aria-hidden="true"
          class="svg-inline--fa fa-link fa-w-16 fa-1x Icon-sc-4jlog9-0 hXLKsx"
          color="var(--color-lapis-lazuli)"
          data-icon="link"
          data-prefix="fas"
          focusable="false"
          role="img"
          viewBox="0 0 512 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M326.612 185.391c59.747 59.809 58.927 155.698.36 214.59-.11.12-.24.25-.36.37l-67.2 67.2c-59.27 59.27-155.699 59.262-214.96 0-59.27-59.26-59.27-155.7 0-214.96l37.106-37.106c9.84-9.84 26.786-3.3 27.294 10.606.648 17.722 3.826 35.527 9.69 52.721 1.986 5.822.567 12.262-3.783 16.612l-13.087 13.087c-28.026 28.026-28.905 73.66-1.155 101.96 28.024 28.579 74.086 28.749 102.325.51l67.2-67.19c28.191-28.191 28.073-73.757 0-101.83-3.701-3.694-7.429-6.564-10.341-8.569a16.037 16.037 0 0 1-6.947-12.606c-.396-10.567 3.348-21.456 11.698-29.806l21.054-21.055c5.521-5.521 14.182-6.199 20.584-1.731a152.482 152.482 0 0 1 20.522 17.197zM467.547 44.449c-59.261-59.262-155.69-59.27-214.96 0l-67.2 67.2c-.12.12-.25.25-.36.37-58.566 58.892-59.387 154.781.36 214.59a152.454 152.454 0 0 0 20.521 17.196c6.402 4.468 15.064 3.789 20.584-1.731l21.054-21.055c8.35-8.35 12.094-19.239 11.698-29.806a16.037 16.037 0 0 0-6.947-12.606c-2.912-2.005-6.64-4.875-10.341-8.569-28.073-28.073-28.191-73.639 0-101.83l67.2-67.19c28.239-28.239 74.3-28.069 102.325.51 27.75 28.3 26.872 73.934-1.155 101.96l-13.087 13.087c-4.35 4.35-5.769 10.79-3.783 16.612 5.864 17.194 9.042 34.999 9.69 52.721.509 13.906 17.454 20.446 27.294 10.606l37.106-37.106c59.271-59.259 59.271-155.699.001-214.959z"
            fill="currentColor"
          />
        </svg>
      </button>
      <button
        class="Button-sc-1bpx84u-0 eyWUJN css-vhv7r1"
        title="Bouton supprimant A Reference Type \\"An Uncategorized Reference\\""
      >
        <svg
          aria-hidden="true"
          class="svg-inline--fa fa-trash fa-w-14 fa-1x Icon-sc-4jlog9-0 hXLKsx"
          color="var(--color-lapis-lazuli)"
          data-icon="trash"
          data-prefix="fas"
          focusable="false"
          role="img"
          viewBox="0 0 448 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M432 32H312l-9.4-18.7A24 24 0 0 0 281.1 0H166.8a23.72 23.72 0 0 0-21.4 13.3L136 32H16A16 16 0 0 0 0 48v32a16 16 0 0 0 16 16h416a16 16 0 0 0 16-16V48a16 16 0 0 0-16-16zM53.2 467a48 48 0 0 0 47.9 45h245.8a48 48 0 0 0 47.9-45L416 128H32z"
            fill="currentColor"
          />
        </svg>
      </button>
    </div>
    <div
      class="css-zkfaav"
    >
      <svg
        aria-hidden="true"
        class="svg-inline--fa fa-circle fa-w-16 fa-sm Icon-sc-4jlog9-0 gomlef"
        color="black"
        data-icon="circle"
        data-prefix="fas"
        focusable="false"
        role="img"
        viewBox="0 0 512 512"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M256 8C119 8 8 119 8 256s111 248 248 248 248-111 248-248S393 8 256 8z"
          fill="currentColor"
        />
      </svg>
      <span
        class="List__Text-sc-1pm6mpm-1 bKJbSY"
      >
        An Agreement Reference
      </span>
      <button
        class="Button-sc-1bpx84u-0 eyWUJN css-vhv7r1"
        title="Bouton supprimant A Reference Type \\"An Agreement Reference\\""
      >
        <svg
          aria-hidden="true"
          class="svg-inline--fa fa-trash fa-w-14 fa-1x Icon-sc-4jlog9-0 hXLKsx"
          color="var(--color-lapis-lazuli)"
          data-icon="trash"
          data-prefix="fas"
          focusable="false"
          role="img"
          viewBox="0 0 448 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M432 32H312l-9.4-18.7A24 24 0 0 0 281.1 0H166.8a23.72 23.72 0 0 0-21.4 13.3L136 32H16A16 16 0 0 0 0 48v32a16 16 0 0 0 16 16h416a16 16 0 0 0 16-16V48a16 16 0 0 0-16-16zM53.2 467a48 48 0 0 0 47.9 45h245.8a48 48 0 0 0 47.9-45L416 128H32z"
            fill="currentColor"
          />
        </svg>
      </button>
    </div>
  </div>
</div>
`;
